### DISCUSSION TITLE
Maximum show rule depth exceeded

### DISCUSSION THREAD

**Discussion by lettuceDestroyer** [typst/typst#4080] [Q&A]
I'm trying to edit table cells based on their value. When the value equals "succeeded" the table cell should be green and when the value equals "failed" it should be red.

What I tried is this:
```typst
show figure.where(kind: table) : it => {
    show table.cell: it => {
      if(it.body == [succeeded]){
        table.cell(it.body, fill: green, inset: it.inset)
      }else if(it.body == [failed]){
        //throws: maximum show rule depth exceeded
        table.cell(it.body, fill: red, inset: it .inset)
      }
      else{
        it
      }
    }
    it.body
    v(it.gap, weak: true)
    it.caption
  }
```

And this:
```typst
  show figure.where(kind: table) : it => {
    if(it.caption.body.has("text")){
      if it.caption.body.text.contains("Testing") {
        show table.cell: it => {
          if(it.body == [succeeded]){
            table.cell(it.body, fill: green, inset: it.inset)
          }else if(it.body == [failed]){
            //throws: maximum show rule depth exceeded
            table.cell(it.body, fill: red, inset: it.inset)
          }
          else{
            it
          }
        }
        it.body
        v(it.gap, weak: true)
        it.caption
      }
    }
  }
```

The reason why I do it like this and not like the [official Typst documentation](https://typst.app/docs/reference/model/table/) is because I'm trying to separate the content and styling.

**Comment by l0wx** [comment]
1. The `show` rule nesting issue known with a planned feature for resolution called Revocation, cf. https://discord.com/channels/1054443721975922748/1175894504146993304/1216194203680378970
2. For many cases, there's a workaround `a la:
```typst
#show cell.where(fill: {}): it=> …// When `fill:` is set for the returned `cell()`, this `show` rule doesn't match
```
3. There seems to be a bug that leads to the `fill:` argument of `cell()` getting disobeyed in `show cell` rules.
4. Anyhow, the more idiomatic way theoretically would be alike:
```typst
#show cell.where(body: [succeeded]): set cell(fill: green)
```
3. There seems to be another bug leading to `show cell: set cell` getting disobeyed altogether albeit that's being the go-to approach for other elements:
```typst
#show rect: set rect(fill: blue)
```
>   […]I'm trying to separate the content and styling.
4. When you are processing structured data (like your repeating keywords à la `failed` are suggesting), directly utilizing `table()`/`grid()` is unexpedient anyway, as lossy conversion to `content` happens before `set`/`show` rules format. What about some `content`ualization wrapper function like the following instead:
```typst
#let tabularize-outcomes(..args) = table(
    columns: 2 * (1fr,),
    header[Affair][Outcome],
    ..for (aff, outc) in args.pos() {(
        aff,
        if true == outc {cell(
            fill: green,
            "succeeded",
        )} else if false == outc {cell(
            fill: red,
            "failed",
        )} else if {} == outc {cell(
            fill: gray,
            "unknown",
        )} else {outc},
    )},
)
```
